src/components/Carousel/Carousel.js

```diff
@@ -0,0 +1,93 @@
+import React, { useState, useEffect } from "react";
+import "./Carousel.scss";
+
+const Carousel = () => {
+  const [curIndex, setCurIndex] = useState(1);
+  const [action, setAction] = useState(1);
+  const [isTransition, setIsTransition] = useState(false);
+
+  useEffect(() => {
+    const timer = setInterval(() => setCurIndex(curIndex + 1), 4000);
+    return () => {
+      clearInterval(timer);
+    };
+  });
+
+  useEffect(() => {
+    setTimeout(() => {
+      console.log("hi");
+      setIsTransition(false);
+    }, 1000);
+    // setTimeout(() => {
+    //   setAction(0);
+    // }, 500);
+    curIndex === 0 &&
+      setTimeout(() => {
+        setAction(0);
+        setCurIndex(3);
+        setTimeout(() => {
+          setAction(1);
+        }, 20);
+      }, 1000);
+    curIndex === 4 &&
```

Comment:
```suggestion
    curIndex === imageList.length-1 &&
```
* 마지막 인덱스가 됐을 때 의 조건이라면, 이미지 배열에 마지막으로 접근하는게 좋겠습니다.
* 이미지가 늘어다면 고정된 4가 아닌, 다른 숫자가 들어올 수도 있으니까요

Reply:
적용하였습니다!